src/native/libs/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -416,12 +416,12 @@ PAL_TlsIo AppleCryptoNative_SslRead(SSLContextRef sslContext, uint8_t* buf, uint
     return OSStatusToPAL_TlsIo(status);
 }
 
-int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
+int32_t AppleCryptoNative_SslIsHostnameMatch(SecTrustRef existingTrust, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
```

Comment:
I have put up https://github.com/dotnet/runtime/pull/117611 so non-network-framework changes in this PR will eventually be reverted (if that PR gets merged)